modules/reactor/include/meshgenerators/HexagonMeshTrimmer.h

```diff
@@ -0,0 +1,158 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+#include "PolygonMeshGeneratorBase.h"
+#include "MooseEnum.h"
+
+/**
+ * This HexagonMeshTrimmer object takes in a polygon/hexagon concentric circle mesh and
+ * renames blocks on a user-defined azimuthal segment / wedge of the mesh.
+ */
+class HexagonMeshTrimmer : public PolygonMeshGeneratorBase
+{
+public:
+  static InputParameters validParams();
+
+  HexagonMeshTrimmer(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+protected:
+  /// Input mesh to be modified
+  const MeshGeneratorName _input_name;
+  /// Peripheral regions that need to be trimmed
+  std::vector<unsigned short> _trim_peripheral_region;
+  /// Name of the section formed by peripheral trimming
+  const BoundaryName _peripheral_trimming_section_boundary;
+  /// Sector number to remain
+  const unsigned int _center_trim_sector_number;
+  /// Starting Sector to remain after
+  const unsigned int _trimming_start_sector;
+  /// Name of the section formed by center trimming
+  const BoundaryName _center_trimming_section_boundary;
+  /// Name/ID of external boundary
+  const BoundaryName _external_boundary_name;
+  /// SubdomainName suffix used to rename the converted triangular elements
+  const SubdomainName _tri_elem_subdomain_name_suffix;
+  /// Customized id shift to define subdomain ids of the converted triangular elements
+  const subdomain_id_type _tri_elem_subdomain_shift;
```

Comment:
having a shift seems less convenient than:
- specify id directly
- if unspecified, use next available using that new API that I think Shikhar added

Reply:
The users do not necessarily know whether or how many tri-element subdomains will be generated due to trimming. So shift would be a more convenient approach. If this is not specified, max_subdomain_id will be used as the shift to ensure uniqueness.